src/Mod/Part/Gui/SectionCutting.cpp

```diff
@@ -0,0 +1,1475 @@
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+# include <Inventor/actions/SoGetBoundingBoxAction.h>
+# include <Inventor/nodes/SoCamera.h>
+# include <Inventor/nodes/SoClipPlane.h>
+# include <Inventor/nodes/SoGroup.h>
+# include <Inventor/nodes/SoOrthographicCamera.h>
+# include <Inventor/sensors/SoTimerSensor.h>
+# include <QDialog>
+# include <QDockWidget>
+# include <QPointer>
+//# include <cmath>
+#endif
+
+#include "SectionCutting.h"
+#include "ui_SectionCutting.h"
+#include <App/Link.h>
+#include <App/Part.h>
+#include <Base/UnitsApi.h>
+#include <Gui/Application.h>
+#include <Gui/Command.h>
+#include <Gui/DockWindowManager.h>
+#include <Gui/Document.h>
+#include <Gui/MainWindow.h>
+#include <Gui/View3DInventor.h>
+#include <Gui/View3DInventorViewer.h>
+#include <Mod/Part/App/FeatureCompound.h>
+#include <Mod/Part/App/FeaturePartBox.h>
+#include <Mod/Part/App/FeaturePartCommon.h>
+#include <Mod/Part/App/FeaturePartCut.h>
+#include <Mod/Part/App/FeaturePartFuse.h>
+#include <Mod/Part/App/PartFeatures.h>
+
+using namespace PartGui;
+
+enum Refresh : bool
+{
+    notXValue = false,
+    notYValue = false,
+    notZValue = false,
+    notXRange = false,
+    notYRange = false,
+    notZRange = false,
+    XValue = true,
+    YValue = true,
+    ZValue = true,
+    XRange = true,
+    YRange = true,
+    ZRange = true
+};
+
+SectionCut::SectionCut(Gui::View3DInventor* view, QWidget* parent)
+    : QDialog(parent)
+    , ui(new Ui_SectionCut)
+{
+    // create widgets
+    ui->setupUi(this);
+    ui->cutX->setRange(-INT_MAX, INT_MAX);
+    ui->cutY->setRange(-INT_MAX, INT_MAX);
+    ui->cutZ->setRange(-INT_MAX, INT_MAX);
+
+    view = view;
+    Gui::View3DInventorViewer* viewer = view->getViewer();
+
+    // get all objects in the document
+    doc = Gui::Application::Instance->activeDocument()->getDocument();
+    if (!doc) {
+        Base::Console().Error("SectionCut error: there is no document\n");
+        return;
+    }
+
+    std::vector<App::DocumentObject*> ObjectsList = doc->getObjects();
+    if (ObjectsList.empty()) {
+        Base::Console().Error("SectionCut error: there are no objects in the document\n");
+        return;
+    }
+    // empty the ObjectsListVisible
+    ObjectsListVisible.clear();

```

Comment:
ObjectsListVisible cannot be non-empty here.

Reply:
I don't understand. It must be cleared here, otherwise it won't work.

Reply:
We are inside the constructor of SectionCut and at this point no member function will be called to modify the class member ObjectsListVisible. Since it's empty by default how can it become non-empty here?

Reply:
Ah, my mistake. I thought I have to initialize it explicitly as being empty to assure that there is no garbage in it when it is not initialized.
Since it already is initialized as empty, I can omit this.
(seems I programmed recently too much in FreePascal ;-) )

Reply:
done in #6360